src/ex4/client/src/components/InputComp.jsx

```diff
@@ -0,0 +1,48 @@
+import "../App.css";
+import "../style/InputComp.css";
+import React, { useCallback, useRef, useState } from "react";
+import ButtonComp from "./ButtonComp";
+import plusImg from "../images/plus.png";
+import {} from "../item_client";
+
+function InputComp({ addItem }) {
+	const [inputValue, setInputValue] = useState("");
+	const inputElement = useRef();
+
+	const addItemHandeler = useCallback(() => {
+		addItem(inputValue);
+		inputElement.current.value = "";
+		inputElement.current.focus();
+	}, [addItem, inputValue, inputElement]);
+
+	const eventListenerForEnter = useCallback(
+		(e) => {
+			if (inputValue && e.keyCode === 13) {
```

Comment:
because the check for inputValue is only done here, enter can't be used with empty text but the button can be used.